admin/abstract/views.py

```diff
@@ -0,0 +1,49 @@
+from django.views.generic import FormView
+from django.http import HttpResponseNotFound
+from django.shortcuts import render
+
+
+class GuidFormView(FormView):
+    form_class = None
+    template_name = None
+    object_type = None
+
+    def get_guid_object(self):
+        raise NotImplementedError
+
+    def __init__(self):
+        self.guid = None
+        super(GuidFormView, self).__init__()
+
+    def get(self, request, *args, **kwargs):
+        self.guid = request.GET.get('guid', None)
+        if self.guid is not None:
+            try:
+                guid_object = self.get_guid_object()
+            except (AttributeError, TypeError):
+                return HttpResponseNotFound(
```

Comment:
![screen shot 2016-02-01 at 10 45 47 am](https://cloud.githubusercontent.com/assets/5169088/12722072/0e796afa-c8d1-11e5-8411-34bdc50a61a2.png)

Less friendly I think.

Maybe I should go for a signature more like: `admin/nodes/whup2` rather than using the `?guid=`